### ISSUE TITLE
Support other modules on /gen folder

### ISSUE THREAD

**Issue by georgesboris** [dillonkearns/elm-pages#165]
Hey Dillon! I'm having a great time using elm-pages but I came across this issue:

I'm trying to use `elm-pages` alongside `elm-tailwind-modules`. Both these packages generate elm modules inside the `/gen` folder by default. I initially thought that would be great as I would keep all codegen output in this same directory (the directory is listed on elm/json `source_directories`, as expected).

However, apparently elm-pages is not compatible with other modules inside this folder as I keep running into "module not found" issues. I had to change the output folder of elm-tailwind-modules to be able to make this work. I tried both putting into source and into a third folder also included in `source_directories`, both worked. The only problem seems to happen when I add more stuff into `/gen`.

To summarize:

🚫 module defined inside `gen`
✅ module defined inside `gen2`
✅ module defined inside `src`

Cheers! Let me know if you need any help here.

**Comment by sparksp** [comment]
I've just encountered this myself. `elm make src/Main.elm` works fine on the project but `elm-pages build` cannot see the other modules within `./gen`. I don't want to mix these into `./src` although I would be open to considering another gen folder of things that I want to commit. 🤔

**Comment by dillonkearns** [comment]
Thanks to both of you for reporting the issue.

The issue is that the codegen uses a different set of files for the headless CLI app (which is a Platform.worker) and the browser app (which is a Browser.application). So `elm-pages` does need to generate separate files in these two modes. It generates a separate app in a sub-folder and uses a modified version of the apps `elm.json`, but with the `gen` folder removed so it can replace it with separate generated files for the headless app.

So `elm-pages` will need to do something similar, so I think the solution that makes the most sense is probably to just use a dedicated `gen` folder (something like `.elm-pages`). I'll try to get this in for the 2.0 release 👍
